### PR TITLE
Update normalization parameters and add estimator params validation

### PR DESCRIPTION
**Pull request by avolkov-intel**
## Description

- Add different normalization options
- Remove implicit normalization from loaders
- Add estimator parameters validation

Estimator parameters validation can be useful in case you want to override some parameter like n_jobs using -p algorithm:estimator_params:n_jobs=64 for the benchmarks. Currently it would fail since some estimators don't support n_jobs and you would need to run the benchmarks separately. In this approach this parameter will be simply ignored by estimators that don't support it and warning will be shown.

---

<details>
<summary>Checklist:</summary>

**Completeness and readability**

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with updates and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

</details>

### CODE REVIEW

File: configs/regular/logreg.json [hunk 1]
```diff
@@ -61,21 +61,24 @@
             {
                 "data": {
                     "dataset": "hepmass",
-                    "split_kwargs": { "train_size": 0.1, "test_size": null }
+                    "split_kwargs": { "train_size": 0.1, "test_size": null },
```
Comment (ethanglaser) [inline configs/regular/logreg.json]: Seems like this is the only case where benchmark behavior changes - is it intended?

File: configs/weekly/dbscan.json [hunk 2]
```diff
@@ -4,14 +4,19 @@
         "high-load dbscan datasets": {
             "data": [
                 {
-                    "dataset": ["cifar", "road_network", "covtype"],
+                    "dataset" : "cifar",
+                    "split_kwargs": { "ignore" : true },
```
Comment (ethanglaser) [inline configs/weekly/dbscan.json]:
```suggestion
                    "dataset": "cifar",
                    "split_kwargs": { "ignore": true },
```

File: configs/regular/svm.json [hunk 3]
```diff
@@ -30,7 +30,7 @@
                 "data": {
                     "dataset": "mnist",
                     "split_kwargs": { "train_size": 20000, "test_size": null },
-                    "preprocessing_kwargs": { "normalize": false }
+                    "preprocessing_kwargs": {"normalize" : null}
```
Comment (ethanglaser) [inline configs/regular/svm.json]:
```suggestion
                    "preprocessing_kwargs": { "normalize": null }
```
Minor adjustment but looks like spaces have been added before colons in a few places throughout the configs

File: configs/regular/logreg.json (hunk 1 above)
Comment (avolkov-intel) [reply configs/regular/logreg.json]: I think it was done for a reason but let me check the convergence for both options

File: configs/common/knn.json [hunk 4]
```diff
@@ -8,7 +8,7 @@
                 }
             },
             "data": {
-                "preprocessing_kwargs": { "normalize": true }
+                "preprocessing_kwargs": { "normalize": "standard" }
```
Comment (david-cortes-intel) [inline configs/common/knn.json:11]: Would the jsons with the datasets be able to override this? For example:
https://github.com/avolkov-intel/scikit-learn_bench/blob/a2a75152d6bae2cf3bcdf350125369b04f289b33/configs/regular/knn.json#L8

It's not clear from the docs how it would work when specified more than once.

File: requirements.txt [hunk 5]
```diff
@@ -6,6 +6,7 @@ pandas
 tabulate
 fastparquet
 h5py
+openml
```
Comment (david-cortes-intel) [inline requirements.txt:9]: Looks like you need to rebase from the current main.

File: sklbench/datasets/common.py [hunk 6]
```diff
@@ -219,9 +224,18 @@ def preprocess_x(
...
+        if normalize == "standard":
+            scaler = StandardScaler(with_mean=True, with_std=True)
+        elif normalize == "mean":
+            scaler = StandardScaler(with_mean=True, with_std=False)
+        elif normalize == "minmax":
+            scaler = MinMaxScaler(feature_range=(0, 1))
+        else:
+            logger.warning(f'Unknown "{normalize}" normalization type.')
+        if scaler is not None:
+            return pd.DataFrame(scaler.fit_transform(x), columns=x.columns, index=x.index)
```
Comment (david-cortes-intel) [inline sklbench/datasets/common.py]: Wouldn't this make it ignore `return_type == np.ndarray`?

File: requirements.txt (hunk 5 above)
Comment (avolkov-intel) [reply requirements.txt:9]: It's rebased I just added openml to general requirements.txt file. Currently it's only added to sklearn specific requirements

File: configs/regular/logreg.json (hunk 1 above)
Comment (avolkov-intel) [reply configs/regular/logreg.json]: Essentially, there's no effect on the result: accuracy and number of iterations stays the same

File: sklbench/datasets/common.py (hunk 6 above)
Comment (avolkov-intel) [reply sklbench/datasets/common.py]: Currently it works correctly for all return_types as intermediate data is always represented in pandas format. However, this conversion is indeed redundant if return_type is not a pandas dataframe

File: configs/common/knn.json (hunk 4 above)
Comment (avolkov-intel) [reply configs/common/knn.json:11]: As I understand the logic of the benchmarks it's possible to override the parameter value on the lower level which is the case for all parameters not only normalization. I double checked this for this concrete case, normalize parameter is overriden. This is documented at the bottom of this file: https://github.com/IntelPython/scikit-learn_bench/tree/main/docs 

I think that's design pattern of the sklbench, if we want to document it more properly it's better to do this in separate PR

File: sklbench/datasets/common.py (hunk 6 above)
Comment (david-cortes-intel) [reply sklbench/datasets/common.py]: Is that because it then goes through train_test_split? Isn't that step optional?

**Review by ethanglaser — APPROVED** [review]
LGTM once David's comments are resolved. We should add a json linter step to CI and unify formatting in sklbench configs.

**Review by david-cortes-intel — APPROVED** [review]
LGTM, but could also change array.values to array.to_numpy().

**Review by david-cortes-intel — APPROVED** [review]
